### PR TITLE
chore: update sqllogictests crate to 0.19.1

### LINKED ISSUE #14874: Update sqllogictest crate to the latest
**Summary**
https://github.com/risinglightdb/sqllogictest-rs

### PR DESCRIPTION
**Pull request by finchxx**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

- Fixes #14874 

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [x] Other (please describe):
update crate version

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15242)

### CODE REVIEW

File: tests/sqllogictests/suites/duckdb/join_small_block/semianti/antijoin.test [hunk 1]
```diff
@@ -38,10 +38,8 @@ SELECT * FROM left_table ANTI JOIN (SELECT a as foo from right_table where b = 1
 43	1	1
 
 # Should throw error when filtering on column in right table
-statement error
+statement error 1065
 SELECT * FROM left_table ANTI JOIN right_table ON left_table.a = right_table.a WHERE right_table.a < 43;
-----
-Binder Error
```
Comment (xudong963) [inline tests/sqllogictests/suites/duckdb/join_small_block/semianti/antijoin.test:44]: it seems the format doesn't make sense: https://github.com/risinglightdb/sqllogictest-rs/releases/tag/v0.18.0

If we don't remove `----`, what will happen?

Comment (xudong963) [reply tests/sqllogictests/suites/duckdb/join_small_block/semianti/antijoin.test:44]: I think showing specific error info is better.

Comment (finchxx) [reply tests/sqllogictests/suites/duckdb/join_small_block/semianti/antijoin.test:44]: > I think showing specific error info is better.

how about inline error? Regular expressions have not been supported in the multiline error yet.

Comment (xudong963) [reply tests/sqllogictests/suites/duckdb/join_small_block/semianti/antijoin.test:44]: make sense

**Review by sundy-li — APPROVED** [review]

File: tests/sqllogictests/suites/duckdb/join/semianti/antijoin.test [hunk 2]
```diff
@@ -35,10 +35,8 @@ SELECT * FROM left_table ANTI JOIN (SELECT a as foo from right_table where b = 1
 43	1	1
 
 # Should throw error when filtering on column in right table
-statement error
+statement error (?s)1065.*column a doesn't exist
```
Comment (xudong963) [inline tests/sqllogictests/suites/duckdb/join/semianti/antijoin.test:38]: What's `(?s)`?

Comment (finchxx) [reply tests/sqllogictests/suites/duckdb/join/semianti/antijoin.test:38]: It is used to fetch the multiline error `column a doesn't exist`. The original error info is as follows:

```
        mysql client error: Server error: `ERROR HY000 (1105): SemanticError. Code: 1065, Text = error: 
  --> SQL:1:98
  |
1 | SELECT * FROM left_table ANTI JOIN right_table ON left_table.a = right_table.a WHERE right_table.a < 43;
  |                                                                                                  ^ column a doesn't exist

.'

```

Comment (xudong963) [reply tests/sqllogictests/suites/duckdb/join/semianti/antijoin.test:38]: oh, it's a regex syntax.

**Review by xudong963 — APPROVED** [review]

**Comment by BohuTANG** [comment]
@finchxx Thanks for your contribution!

**Pull request merged**